docs/ibc/1_IBC_DESIGN_PHILOSOPHY.md

```diff
@@ -0,0 +1,49 @@
+
+### Locality of communication & information
+
+IBC makes no assumptions, and relies upon no characteristics, of the topological structure of the network of blockchains in which it is operating. No view of the global network-of-blockchains topology is required: security & correctness can be reasoned about at the level of a single connection between two chains, and by composition reasoned about for subgraphs in the network topology. Users and chains can reason about their assumptions and risks given information about only part of the network graph of blockchains they know and trust (to variable degrees). There is no necessary "root chain" in IBC — some subgraphs of the global network may evolve into a hub-spoke structure, others may remain tightly connected, others still may take on more exotic topologies.
+
+### Locality of trust & security
+
+Users of IBC — at the blockchain level and at the human or smart contract level — choose which consensus algorithms, state machines, and validator sets they "trust" (to behave in a particular way, e.g. < 1/3 Byzantine) and in which ways they trust them. Assuming the IBC protocol is implemented correctly, users are never exposed to risks of application-level invariant violations (such as asset inflation) due to Byzantine behavior or faulty state machines transitions committed by validator sets or blockchains they did not explicitly decide to trust. This is particulary important in the expected large network topology of interconnected blockchains, where some number of blockchains and validator sets can be expected to be Byzantine occaisionally — IBC, implemented conservatively, bounds the risk and limits the possible damage incurred.
+
+### Locality of permissioning
+
+Actions in IBC — such as opening a connection, creating a channel, or sending a packet — are permissioned locally by the state machines and actors involved in a particular connection between two chains. Individual chains could choose to require approval from a permissioning mechanism (such as governance) for specific application-layer actions (such as delegated-security slashing), but for the base protocol, actions are permissionless (modulo gas & storage costs) — by default, connections can be opened, channels created, and packets sent without any approval process. Of course, users themselves must inspect the state & consensus of each IBC connection and decide whether it is safe to used (based e.g. on the roots-of-trust stored).
```

Comment:
> of course, users themselves must inspect the state & consensus of each IBC connection and decide whether it is safe to used

can this be done automatically?

Reply:
A client-side wallet could have some sort of safety heuristic, if that's what you mean.